AttitudeManager/ControlAlgorithms/Src/AM_FixedControl.cpp

```diff
@@ -1,17 +1,116 @@
+                            pitch_i_windup, -maxPitchAngle, maxPitchAngle};
+    PIDController pid_airspeed{airspeed_kp,       airspeed_ki,  airspeed_kd, 
+                               airspeed_i_windup, airspeed_min, airspeed_max};                        
+
+    // Compute values for mixing
+    float bank = pid_bank.execute(targetBank, currentAttitude.roll, currentAttitude.rollRate);
+
+    float pitch = pid_pitch.execute(targetPitch, currentAttitude.pitch, currentAttitude.pitchRate);
+
+    float rudderSet = rudderPercent(bank);
+    
+    rudderSet = pid_rudder.execute(0.0f, targetHeading);
 
-    // do something
+    float airspeed = pid_airspeed.execute(targetAltitude, currentAttitude.altitude, // handling altitude with airspeed? Define altitude and airspeed seperately?
+                                          currentAttitude.rateOfClimb);
+    
+    float roll = DEG_TO_RAD(bank);
+
+    // mix the PID's
+    float engineOutput = 
+        mixPIDs(configs[Engine].stateMix, roll, pitch, rudderSet, airspeed); 
+    float leftAileronOutput = 
+        mixPIDs(configs[LeftAileron].stateMix, roll, pitch, rudderSet, airspeed);
+    float rightAileronOutput = 
+        mixPIDs(configs[RightAileron].stateMix, roll, pitch, rudderSet, airspeed);
+    float rudder = 
+        mixPIDs(configs[Rudder].stateMix, roll, pitch, rudderSet, airspeed);
+    float elevator = 
+        mixPIDs(configs[Elevator].stateMix, roll, pitch, rudderSet, airspeed);
+
+    // Verify values
+    assert(configs[Engine].channel < outputsLength); // Question: Engine controlled as some percentage? Indicative of airspeed?
+    outputs[configs[Engine].channel] = engineOutput;
+
+    assert(configs[LeftAileron].channel < outputsLength); 
+    outputs[configs[LeftAileron].channel] = leftAileronOutput;
+        
+    assert(configs[RightAileron].channel < outputsLength);
+    outputs[configs[RightAileron].channel] = rightAileronOutput;
+        
+    assert(configs[Rudder].channel < outputsLength);
+    outputs[configs[Rudder].channel] = rudder;
+        
+    assert(configs[Elevator].channel < outputsLength);
+    outputs[configs[Elevator].channel] = elevator;
+
+}
+// Modify for fixed wing?
+float FixedControl::mixPIDs(StateMix actuator, float roll, float pitch, float yaw,
+                  float altitude) const {
+    return constrain<float>(actuator.pitch * pitch + actuator.roll * roll +
+                                actuator.yaw * yaw +
+                                actuator.velocity_z * altitude,
+                            100, 0);
+}
 
-    // return output
-    // ActuatorOutput actuator;
-    // assert(actuator.channel < outputsLength);
-    // outputs[actuator.channel] = actuator.percent;
+float FixedControl::rudderPercent(float bankAngle) const {
+    return((rudder_scaling_factor * bankAngle) / (ZP_PI / 2.0)) * 100.0; // "very simple for now. Experiments may give us a better formula. The PID will fix any discrepancy though"
```

Comment:
not sure what this does...would you happen to know?

Reply:
I think this has to do with how your rudder positions itself in a coordinated turn, though I am not sure. I'll see if I can find anything online.

Reply:
ok, in this instance, I think it's important to keep in mind that our drone could have multiple different sources of yaw control, and we want to be able to mix them in.

I'm also not super sure how coordinated turns works, I'll have to look it up, but I'm more so wondering why we don't have the rudder be assigned in a pid value way to the actuators that can impact our yaw state?